### ISSUE TITLE
Updating kubelet config doesn't mark nodes as NeedsUpdate

### ISSUE THREAD

**Issue by marek-obuchowicz** [kubernetes/kops#3076]
I have edited cluster configuration with `kops edit cluster` - i've added there imageGC settings for kubelet. Afterwards the settings have been updated in config files in s3 bucket. However, settings aren't picked up by already running kubelet and nodes are not marked as `NeedsUpdate`:
```
Using cluster from kubectl context: p1.hidden.xxx

NAME			STATUS	NEEDUPDATE	READY	MIN	MAX	NODES
master-us-east-1a	Ready	0		1	1	1	1
master-us-east-1c	Ready	0		1	1	1	1
master-us-east-1e	Ready	0		1	1	1	1
nodes			Ready	0		3	3	3	3
```

I had to manually change IG settings of all instance groups to "force" mark nodes as NeedsUpdate, which allowed me to run `rolling-update`.

**Comment by gambol99** [comment]
Yeah ... i've noticed this as well. Pretty much any of the componentConfig settings don't appear to show an update is required.

**Comment by gambol99** [comment]
The problem is due to the fact and 'edit' or 'replace' command automatically updates the bucket, so there is nothing to actually to diff the current to updated config. Somewhat of a massive problem if you want to drive from CI ... I think was thinking about this the other day, i think a quick work around this would be to ensure the ig spec and perhaps some of the cluster spec makes it's way into userdata which still hold the previous state .. @justinsb what do you reckon?

**Comment by KashifSaadat** [comment]
/assign
/close

This is now resolved in the latest kops releases (v1.8.0 and up).
